bridge-token-factory/tests/token_transfer.rs

```diff
@@ -63,161 +84,218 @@ impl From<Proof> for AugmentedProof {
-        )
-    };
-    ($signer:expr, $contract:ident.$method:ident($arg:expr), deposit=$deposit:expr) => {
-        call_borsh!(
-            $signer,
-            $contract,
-            $method,
-            $arg,
-            near_sdk_sim::DEFAULT_GAS,
-            $deposit
-        )
+fn err_is(result: &Result<CallExecutionDetails, anyhow::Error>, expected: &str) {
+    let status = match result {
+        Ok(result) => {
+            assert!(!result.is_success(), "Expected error found {:?}", result);
+            format!("{:?}", result.outcome())
+        }
+        Err(err) => {
+            format!("{:?}", err)
+        }
     };
-}
 
-fn err_is(result: &ExecutionResult, expected: &str) {
-    assert!(!result.is_ok(), "Expected error found {:?}", result);
-    let status = format!("{:?}", result.outcome().status);
     assert!(status.contains(expected), "{}", status);
 }
 
 #[test]
 fn test_eth_token_transfer() {
-    let (user, factory) = setup_token_factory();
-
-    let alice = user.create_user(ALICE.to_string(), to_yocto("100"));
-    let factory_id = FACTORY.to_string();
-
-    call!(
-        user,
-        factory.deploy_bridge_token(DAI_ADDRESS.to_string()),
-        deposit = to_yocto("35")
+    let (user, factory, worker) = create_contract();
+    let rt = Runtime::new().unwrap();
+
+    assert!(&rt
+        .block_on(
+            user.call(&worker, factory.id(), "deploy_bridge_token")
+                .deposit(parse_near!("35 N"))
+                .args(
+                    json!({"address": DAI_ADDRESS.to_string()})
+                        .to_string()
+                        .into_bytes()
+                )
+                .max_gas()
+                .transact()
+        )
+        .unwrap()
+        .is_success());
+
+    let mut token_account_id: String = std::str::from_utf8(
+        &rt.block_on(
+            factory.view(
+                &worker,
+                "get_bridge_token_account_id",
+                json!({"address": DAI_ADDRESS.to_string()})
+                    .to_string()
+                    .into_bytes(),
+            ),
+        )
+        .unwrap()
+        .result,
     )
-    .assert_success();
-
-    let token_account_id: String =
-        view!(factory.get_bridge_token_account_id(DAI_ADDRESS.to_string())).unwrap_json();
-    assert_eq!(token_account_id, format!("{}.{}", DAI_ADDRESS, FACTORY));
-
-    let alice_balance: String =
-        call_json!(user, token_account_id.ft_balance_of({"account_id": ALICE.to_string()}))
-            .unwrap_json();
-    assert_eq!(alice_balance, "0");
-
-    let total_supply: String = call_json!(user, token_account_id.ft_total_supply({})).unwrap_json();
-    assert_eq!(total_supply, "0");
+    .unwrap()
+    .to_string();
+    assert_eq!(
+        token_account_id,
+        format!("\"{}.{}\"", DAI_ADDRESS, factory.id())
+    );
+    token_account_id = token_account_id[1..token_account_id.len() - 1]
+        .parse()
+        .unwrap();
```

Comment:
Let's use String methods or serde for it (and maybe a separate function). After this, the previous assert  statement should be without quotes too.

Reply:
fixed: https://github.com/aurora-is-near/rainbow-token-connector/pull/159/commits/98381780c4a98bca3d1a24b1b44d072b8d40e27c